### PR TITLE
Pdf extracts table

### PR DESCRIPTION
**Pull request by cantsin**
Splitting up PDF extracts work so it's not so large; partially addresses https://github.com/18F/FAC-Distiller/issues/67

Changes proposed in this pull request:
- Adds PDFExtract table and migrations
- Updates Pipfile with pdf/spacy requirements
- Regenerates Pipfile.lock 
- Adds (_sigh_) requirements.txt due to language pack requirements
- Updates README
- Minor tweaks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +128,17 @@ pipenv run python manage.py load_table --findingtext
 
 In the deployed environment, `django-apscheduler` is used to refresh all tables daily at 12:00 AM EST.
 
+### requirements.txt
+
+To install the spacy English language pack (`en_core_web_sm`) on cloud.gov, we must explicitly specify this dependency as a file download in `requirements.txt` (see this [related spacy Github issue](https://github.com/explosion/spaCy/issues/3536 "spacy Github issue about installation of language models from pyPI"). Unfortunately, this also means maintaining Pipenv _and_ a requirements file.
```
Comment (danielnaab) [inline README.md]: Could you accomplish the same by putting this in the `Pipfile`?

```ini
en-core-web-sm = {file = "https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.1.0/en_core_web_sm-2.1.0.tar.gz"}
```

File: distiller/data/models/pdf_extract.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    # Map to General/Audit
+    audit = CompositeForeignKey(
+        Audit,
+        on_delete=models.DO_NOTHING,
+        to_fields={
+           'audit_year': 'audit_year',
+           'dbkey': 'dbkey'
+        },
+        related_name='pdf_extracts'
+    )
```
Comment (danielnaab) [inline distiller/data/models/pdf_extract.py]: An index on `(audit_year, dbkey)` should help speed this join up.

File: README.md (hunk 1 above)
Comment (cantsin) [reply README.md]: Yep, we already have this line in the `Pipfile` (although for version 2.2.0). And this doesn't work: I think cloud.gov generates the requirements.txt itself if the file doesn't exist, and the resulting requirements.txt is not set up correctly. Maybe there's a solution if we tweak the buildpack but that is a bit too far down the rabbit hole for me...

Here's what happens if I push without `requirements.txt`.

```bash
$ rm requirements.txt
$ cf push
.... a lot of text ... 
          Generating 'requirements.txt' from Pipfile.lock
... more text ...
            ERROR: Could not find a version that satisfies the requirement en-core-web-sm (from -r /tmp/app/requirements.txt (line 35)) (from versions: none)
          ERROR: No matching distribution found for en-core-web-sm (from -r /tmp/app/requirements.txt (line 35))
   Failed to compile droplet: Failed to run all supply scripts: exit status 14
   Exit status 223
```

File: distiller/data/models/pdf_extract.py (hunk 2 above)
Comment (cantsin) [reply distiller/data/models/pdf_extract.py]: Absolutely right, thanks for the catch. Will add this.

File: README.md (hunk 1 above)
Comment (danielnaab) [reply README.md]: Huh, weird one 🤕

**Pull request merged**